### ISSUE TITLE
Can't publish crate with dashu dependency

### ISSUE THREAD

**Issue by mthom** [cmpute/dashu#33]
I'm trying to publish a crate that depends on the forked dashu at github.com/coasys/dashu but `cargo publish --dry-run` complains that it can't find the functions associated with the `ToPrimitive` trait of `num_traits`. Is there a workaround for this? Thanks.

**Comment by cmpute** [comment]
What did you mean by "depends on the forked dashu at github.com/coasys/dashu" please? Can you elaborate on your Cargo.toml and error information?

**Comment by mthom** [comment]
Yes. I'm using the dashu at the named repo like this in my Cargo.toml file:

`dashu = { git = "https://github.com/coasys/dashu.git", version = 0.3.1 }`

Ordinarily this works fine, but when I try to compile the crate for publishing on crates.io (using `cargo publish --dry-run`) I get:

```
error[E0599]: no method named `to_i64` found for struct `arena::TypedArenaPtr<IBig>` in the current scope
   --> src/arithmetic.rs:373:32
    |
373 |             if let Some(n) = i.to_i64() {
    |                                ^^^^^^
    |
   ::: src/arena.rs:188:1
```
and similar errors, all related to the `ToPrimitive` trait not being found. Some recommend to add `use num_traits::cast::ToPrimitive;` but this doesn't work.

**Comment by cmpute** [comment]
IIRC, you can't publish a crate with git dependency, because cargo will try to pull `dashu` from crates.io, which is the version published by me. If you want to publish your crate with `dashu`, please use the official repo.

I noticed that the fork you are using added conversion functionalities like the `to_i64()` in your code, this conversion is implemented as `impl TryFrom<IBig>` in the main branch, so you can use `i.try_into()` instead with the official `dashu` crate.

**Comment by cmpute** [comment]
The issue should be solved by now.

**Comment by mthom** [comment]
I'm afraid it's not. The latest publish dry run produces this batch of errors:

**Comment by cmpute** [comment]
I tried locally and the crate can be published with `dashu` v0.4. I will create a PR tomorrow with additional modifications about more idiomatic usage of `dashu`.
